### PR TITLE
handle amwalPreCheckoutTriggerError

### PR DESCRIPTION
**Pull request by sameh-amwal**

### CODE REVIEW

**Review by Skullsneeze — CHANGES_REQUESTED** [review]
I would use the existing amwalErrorHandler to handle the error. If you've chosen to not use it because you don't want to execute certain logic (eg. add the error message to the magento message system) I would suggest adding a parameter to optionally execute that piece of logic.

The reasoning behind this is that introducing these duplicated code blocks means that if something needs to be changes in how the event is sent, you will need to adjust multiple pieces of code across multiple files, as opposed to one single file

**Comment by sameh-amwal** [comment]
> I would use the existing amwalErrorHandler to handle the error. If you've chosen to not use it because you don't want to execute certain logic (eg. add the error message to the magento message system) I would suggest adding a parameter to optionally execute that piece of logic.
> 
> The reasoning behind this is that introducing these duplicated code blocks means that if something needs to be changes in how the event is sent, you will need to adjust multiple pieces of code across multiple files, as opposed to one single file

makes sense. I added an eventName option to the error handler

File: view/frontend/web/js/checkout-button-product.js [hunk 1]
```diff
@@ -198,6 +198,12 @@ function ($, Component, idsResolver, productInfoResolver, _) {
                         'form': $form,
                         'response': res
                     });
+
+                    self.$checkoutButton.dispatchEvent(new CustomEvent('amwalPreCheckoutTriggerError', {
+                        detail: {
+                            description: res?.responseJSON?.message
+                        }
+                    }));
```
Comment (Skullsneeze) [inline view/frontend/web/js/checkout-button-product.js:206]: If you include `amwalErrorHandler` in this component as well you can use the error handler here too

```suggestion
                    amwalErrorHandler.process(self.checkoutButton, res?.responseJSON?.message, 'amwalPreCheckoutTriggerError');
```

**Review by Skullsneeze — CHANGES_REQUESTED** [review]

File: view/frontend/web/js/checkout-button-product.js (hunk 1 above)
Comment (sameh-amwal) [reply view/frontend/web/js/checkout-button-product.js]: not sure if I did everything correctly, but I did it similar to the other file. Is there a documentation on this Javascript framework?

**Review by Skullsneeze — APPROVED** [review]

**Pull request merged**